pkg/k8s/utils/utils_test.go

```diff
@@ -72,3 +76,219 @@ func TestServiceProxyName(t *testing.T) {
+				PodIPs: []slim_corev1.PodIP{
+					{
+						IP: "10.0.0.1",
+					},
+					{
+						IP: "127.0.0.2",
+					},
+					{
+						IP: "127.0.0.3",
+					},
+				},
+			},
+			want: []string{"10.0.0.1", "127.0.0.2", "127.0.0.3"},
+		},
+		{
+			name: "have empty pod ip",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				PodIPs: []slim_corev1.PodIP{
+					{
+						IP: "127.0.0.2",
+					},
+					{
+						IP: "",
+					},
+				},
+			},
+			want: []string{"127.0.0.2"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ValidIPs(tt.args); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ValidIPs() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIsPodRunning(t *testing.T) {
+	tests := []struct {
+		name string
+		args slim_corev1.PodStatus
+		want bool
+	}{
+		{
+			name: "Pod is not Running",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				Phase:  "Succeeded",
+			},
+			want: false,
+		},
+		{
+			name: "Pod is Running",
+			args: slim_corev1.PodStatus{
+				HostIP: "127.0.0.1",
+				Phase:  "Running",
+			},
+			want: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsPodRunning(tt.args); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("TestIsPodRunning() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetLatestPodReadiness(t *testing.T) {
+	podReadyconditiontrue := slim_corev1.PodCondition{
+		Type:               slim_corev1.PodReady,
+		Status:             slim_corev1.ConditionTrue,
+		LastProbeTime:      slim_metav1.Now(),
+		LastTransitionTime: slim_metav1.Now(),
+		Reason:             "",
+		Message:            "",
+	}
+	podReadyconditionfalse := slim_corev1.PodCondition{
+		Type:               slim_corev1.PodReady,
+		Status:             slim_corev1.ConditionFalse,
+		LastProbeTime:      slim_metav1.Now(),
+		LastTransitionTime: slim_metav1.Now(),
+		Reason:             "",
+		Message:            "",
+	}
+	podReadyconditionUnknown := slim_corev1.PodCondition{
+		Type:               slim_corev1.PodReady,
+		Status:             slim_corev1.ConditionUnknown,
+		LastProbeTime:      slim_metav1.Now(),
+		LastTransitionTime: slim_metav1.Now(),
+		Reason:             "",
+		Message:            "",
+	}
+	podScheduled := slim_corev1.PodCondition{
+		Type:               slim_corev1.PodScheduled,
+		Status:             slim_corev1.ConditionTrue,
+		LastProbeTime:      slim_metav1.Now(),
+		LastTransitionTime: slim_metav1.Now(),
+		Reason:             "",
+		Message:            "",
+	}
+	podContainersReady := slim_corev1.PodCondition{
+		Type:               slim_corev1.ContainersReady,
+		Status:             slim_corev1.ConditionFalse,
+		LastProbeTime:      slim_metav1.Now(),
+		LastTransitionTime: slim_metav1.Now(),
+		Reason:             "",
+		Message:            "",
+	}
+	tests := []struct {
+		name string
+		args slim_corev1.PodStatus
+		want slim_corev1.ConditionStatus
+	}{
+		{
```

Comment:
Can you add some test cases - ideally taken from a running cluster - where pods have multiple Conditions? Because that would be more realistic.

Reply:
Updated.